### DISCUSSION TITLE
Pylance doesn't work if I want to extract RegionProperties from skimage library

### DISCUSSION THREAD

**Discussion by robsver** [microsoft/pylance-release#4080] [Q&A]
Hello everyone, direct and to the point:

```
from skimage.measure import regionprops
properties = regionprops(label)[0]
```

When I type something like `properties.num_pixels` in order to extract the number of foreground pixels from the homonymous variable, Pylance doesn't propose `num_pixels`. The same happens with all the other properties given in the doc as well ([https://scikit-image.org/docs/stable/api/skimage.measure.html#regionprops](url)).

I'm not sure if this occurs with other similar libraries. 
Is there anything I can do ?

I'm using a conda environment with Python 3.9.16.

**Comment by rchiodo** [comment]
What happens when you goto definition on regionprops? My guess is the type information or type stub for regionprops doesn't have a return type. So the type of `properties` cannot be computed.

**Reply by erictraut** [reply]
The `scikit-image` library is an untyped library, i.e. it contains no type annotations. In this case, pylance will attempt to infer types from the source code, but there are limits to what information can be inferred. If this library is important to you, please contact the library maintainers and request that they add type information. Many popular Python libraries have been adding type information to improve the programming experience.

**Reply by robsver** [reply]
Hi @rchiodo & @erictraut, thanks for answering. I understand the problem. In fact it works if I edit regionprops return type. 
It would be helpful to also include suggestions obtained from reading the body of the different functions.
